infra/feast-operator/internal/controller/services/repo_config.go

```diff
@@ -44,71 +45,154 @@ func (feast *FeastServices) getServiceFeatureStoreYaml(feastType FeastServiceTyp
+				if err != nil {
+					return repoConfig, err
 				}
 			}
 		case RegistryFeastType:
 			// Registry server only has a `registry` section
-			if isLocalRegistry {
-				path := DefaultRegistryEphemeralPath
-				var s3AdditionalKwargs *map[string]string
-				if services != nil && services.Registry != nil && services.Registry.Local != nil &&
-					services.Registry.Local.Persistence != nil && services.Registry.Local.Persistence.FilePersistence != nil {
-					filePersistence := services.Registry.Local.Persistence.FilePersistence
-					path = getActualPath(filePersistence.Path, filePersistence.PvcConfig)
-					s3AdditionalKwargs = filePersistence.S3AdditionalKwargs
-				}
-				repoConfig.Registry = RegistryConfig{
-					RegistryType:       RegistryFileConfigType,
-					Path:               path,
-					S3AdditionalKwargs: s3AdditionalKwargs,
+			if isLocalReg {
+				err := setRepoConfigRegistry(services, secretExtractionFunc, &repoConfig)
+				if err != nil {
+					return repoConfig, err
 				}
-				repoConfig.OfflineStore = OfflineStoreConfig{}
-				repoConfig.OnlineStore = OnlineStoreConfig{}
 			}
 		}
 	}
 
 	return repoConfig, nil
 }
 
+func setRepoConfigRegistry(services *feastdevv1alpha1.FeatureStoreServices, secretExtractionFunc func(secretRef string, secretKeyName string) (map[string]interface{}, error), repoConfig *RepoConfig) error {
+	var parametersMap map[string]interface{}
+	filePath := DefaultRegistryEphemeralPath
+	var s3AdditionalKwargs *map[string]string
+	var persistenceType string
+	if services != nil && services.Registry != nil && services.Registry.Local != nil &&
+		services.Registry.Local.Persistence != nil && services.Registry.Local.Persistence.FilePersistence != nil {
+		persistenceType = string(RegistryFileConfigType)
+		filePersistence := services.Registry.Local.Persistence.FilePersistence
+		filePath = getActualPath(filePersistence.Path, filePersistence.PvcConfig)
+		s3AdditionalKwargs = filePersistence.S3AdditionalKwargs
+	} else if services != nil && services.Registry != nil && services.Registry.Local != nil &&
+		services.Registry.Local.Persistence != nil && services.Registry.Local.Persistence.DBPersistence != nil {
+		if len(services.Registry.Local.Persistence.DBPersistence.Type) > 0 {
+			var err error
+			filePath = ""
+			persistenceType = services.Registry.Local.Persistence.DBPersistence.Type
+			secretKeyName := services.Registry.Local.Persistence.DBPersistence.SecretKeyName
+			if len(secretKeyName) == 0 {
+				secretKeyName = persistenceType
+			}
+			parametersMap, err = secretExtractionFunc(services.Registry.Local.Persistence.DBPersistence.SecretRef.Name, secretKeyName)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	repoConfig.Registry = RegistryConfig{
+		RegistryType:       RegistryConfigType(persistenceType),
+		Path:               filePath,
+		S3AdditionalKwargs: s3AdditionalKwargs,
+		DBParameters:       parametersMap,
+	}
```

Comment:
have you tested what happens if a user specificies a conflicting `registry_type` in their secret? we need to be sure the operator has final say on setting that field in the `feature_store.yaml`. Same goes for `type` in online/offlineStores. If you find it can be overwritten by the `parametersMap`, maybe we set `RegistryType` last? As a secondary line of code after this RegistryConfig declaration?

Reply:
maybe the same needs to be done for `filePath` and `s3AdditionalKwargs` ?

Reply:
and some unit tests around this phenomenon? :)

Reply:
maybe test how setting a different `registry_type` in `DBParameters` map will effect the resulting yaml

Reply:
Tracked new GH issue with leftovers